### ISSUE TITLE
HttpServer does not decode gzip "Content-Encoding" requests

### ISSUE THREAD

**Issue by erichwang** [airlift/airlift#335]
HttpServer properly encodes responses to gzip with the "Accept-Encoding" header, but does not decode requests with the gzip "Content-Encoding" header.

curl -v -X POST -T [local_file_name.gz] -H 'Content-Encoding: gzip'

**Comment by wendigo** [comment]
@yuanlihan @erichwang I don't think that Jetty should be responsible for decoding requests. This is a responsibility of different layer - for HTTP server this is only a byte stream

**Comment by yuanlihan** [comment]
@wendigo Thanks for your comment! I got your point. But I think it would make sense to support handling gzip content in [JsonResponseHandler](https://github.com/airlift/airlift/blob/master/http-client/src/main/java/io/airlift/http/client/JsonResponseHandler.java), right?
